index.js

```diff
@@ -28,33 +34,40 @@ async function configure ({ testnet, advanced }) {
     type: 'input',
     name: 'name',
     message: 'Name to assign to this channel:',
-    default: ''
+    default: base64url(crypto.randomBytes(32))
   }]
   for (const field of fields) {
     res[field.name] = (await inquirer.prompt(field))[field.name]
   }
 
+  if (res.provider.toLowerCase() !== 'web3') {
+    throw new Error('This provider is not supported.')
+  }
+
+  // create btp server uri for upstream
   const btpName = res.name || ''
   const btpSecret = hmac(hmac(parentBtpHmacKey, res.parent + btpName), res.provider).toString('hex')
```

Comment:
I think some secret or randomness needs to included in the hash to generate `btpSecret`. The provider is pretty easy to guess (probably Infura mainnet), which means the only source of randomness is the name, but it's not clear to the user that it should be secret/protected (I see by default it generates a random name, but the problem is if they name it themselves). It might not be too hard to run a brute force attack against a connector to guess their token.